File: ConsoleApplication/reader/ClientSocket.cs
using System.Net;
using System.Net.Sockets;

namespace Network
{
    internal class ClientSocket
    {
        internal void ConnectTo(string host, int port)
        {
            mSocket = new Socket(
                SocketType.Stream, ProtocolType.Tcp);

            // connect to a client
            mSocket.Connect(host, port);
        }

        internal int Send(byte[] buffer)
        {
            if (buffer == null)
                return -1;

            return mSocket.Send(buffer);
        }

        Socket mSocket;
    }
}

File: merge/ClientSocket.java
package com.codice.semanticmerge.examples.network;

import java.io.*;

class ClientSocket {
    
    java.net.Socket socket;
    
    int send(byte[] buffer) {
        try {
            OutputStream outStream = socket.getOutputStream();
            outStream.write(buffer);
        } catch (Exception e) {
            e.printStackTrace();
        }
    }
    
    void connectTo(String addr, int port) throws Exception {
        // connect to a client
        socket = new java.net.Socket(addr, port);
    }
}